api/NetUtils.go

```diff
@@ -0,0 +1,34 @@
+package api
+
+import (
+	"strconv"
+	"encoding/json"
+	"net/http"
+	"io/ioutil"
+	"errors"
+)
+
+type StringResult struct {
+	Result string
+	Err error
+}
+
+// getBodyJson takes an http.Response pointer and reads all of its data into a map
+func getBodyJson(response *http.Response) (map[string] string, error) {
+	bodyBytes, err := ioutil.ReadAll(response.Body)
+	
+	if (err != nil) {
+		return nil, err
+	}
+	
+	bodyMap := make(map[string] string)
+
+	json.Unmarshal(bodyBytes, &bodyMap)

```

Comment:
anything that returns an error should be handled. it gets annoying cause its so boiler plate but having
```
err := f()
if err != nil {
  // do something
 // maybe just return err
}
```

is proper because it provides for more robust code. 
theres been interesting conversation relating to error handling in go and some are proposals for [Go 2](https://go.googlesource.com/proposal/+/master/design/go2draft-error-handling-overview.md)